release_dashboard/lib/widgets/create_release_substeps.dart

```diff
@@ -33,10 +55,36 @@ class CreateReleaseSubsteps extends StatefulWidget {
 }
 
 class CreateReleaseSubstepsState extends State<CreateReleaseSubsteps> {
-  // Initialize a public state so it could be accessed in the test file.
   @visibleForTesting
   late Map<String, String?> releaseData = <String, String?>{};
 
+  Object? _error;
+  bool _isLoading = false;
+
+  /// Executes the [run] function to initialize a release.
+  ///
+  /// If there is a [startContext] provided as a parameter, e.g. the fake [startContext] in the test,
+  /// executes the [run] method of that [startContext] instead, otherwise use the [conductor]'s.
+  Future<void> runCreateRelease() {
+    if (widget.startContext != null) {
+      return widget.startContext!.run();
+    }
+    // data captured by the input forms and dropdowns are transformed to conform the formats of StartContext
```

Comment:
why do we have two branches here?

Reply:
Yeah, so in `create_release_substeps_test.dart`, a fake `startContext` is passed as a parameter of the `CreateReleaseSubsteps` widget. [Here](https://github.com/Yugue/cocoon/blob/5494852519fe5fc811d05665781f0c9f503d9d5b/release_dashboard/test/widgets/create_release_substeps_test.dart#L164) is an example. This makes sure that when there is a fake `startContext` passed, that one gets executed instead. 
Currently, only the test uses that parameter. Should I rename the parameter `startContext` to `fakeStartContext`?

Reply:
What you should do is instead provide a `FakeConductorService`. Overall, you should try to avoid functions/constructors having special test-only override parameters (this makes the code more difficult to reason about). This function, which has essentially two code paths, one only for tests the other for real use, is a code smell. It means that the tests aren't really testing your code. I recommend you remove the `StartContext` from the widget, and always call `conductor.createRelease()`

Reply:
Ok yea, that sounds a really good idea. Changed!